pkg/operators/wasm/datasource.go

```diff
@@ -0,0 +1,297 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero"
+	wapi "github.com/tetratelabs/wazero/api"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+)
+
+func (i *wasmOperatorInstance) addDataSourceFuncs(env wazero.HostModuleBuilder) {
+	env.NewFunctionBuilder().
+		WithGoModuleFunction(
+			wapi.GoModuleFunc(i.newDataSource),
+			[]wapi.ValueType{wapi.ValueTypeI64}, // DataSourceName
+			[]wapi.ValueType{wapi.ValueTypeI32}, // DataSource
+		).
+		Export("newDataSource")
```

Comment:
Could this be wrapped? E.g. a function that accepts a string (function name), the wapi.GoModuleFunc,  and the two arrays? Could look cleaner that way.